### PR TITLE
added robots.txt for API endpoints

### LINKED ISSUE #3060: Define a `robots.txt` for the API
## Problem

A not insignificant amount of bot traffic on the API is from web crawlers by search engines, like Yandex for example.

## Description

The API should have a `robots.txt` file to discourage crawlers.

### PR DESCRIPTION
**Pull request by MysteryManav**
## Fixes
- Added robots.txt file

Fixes #3060 by @dhruvkb 

## Description
- Added robots.txt file to the api/templates folder

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

@dhruvkb @obulat

### CODE REVIEW

**Review by dhruvkb — CHANGES_REQUESTED** [review]
@MysteryManav you should also add [your previous changes to `urls/__init__.py`](https://github.com/WordPress/openverse/pull/3148/files#diff-98ae75d77da099c862b0910e3a8f2bce0fc2cce638bee3ccd57224db73c7b87e) to this PR.

Also you can update the PR by pushing more commits to the same branch, you don't need to close your existing PR and create a separate PR for each change.

**Comment by MysteryManav** [comment]
@dhruvkb done, sorry i had to clone the repo again due to some problems on my side, so i forgot to redo the changes, should be fine now

File: api/conf/urls/__init__.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 from django.contrib import admin
 from django.urls import include, path
 from django.views.generic import RedirectView
+from django.views.generic.base import TemplateView
```
Comment (dhruvkb) [inline api/conf/urls/__init__.py:12]: It's possible to combine the two imports.
```suggestion
from django.views.generic import RedirectView, TemplateView
```

**Review by dhruvkb — APPROVED** [review]
LGTM, except for a minor nit.

File: api/conf/urls/__init__.py [hunk 2]
```diff
@@ -35,6 +35,13 @@
     path("admin/", admin.site.urls),
     path("healthcheck/", HealthCheck.as_view(), name="health"),
     path("v1/", include(versioned_paths)),
+    path(
+        "robots.txt/",
```
Comment (dhruvkb) [inline api/conf/urls/__init__.py:39]: I don't know if there should be a trailing slash for a `.txt` file. I'm leaning towards no, there shouldn't be one.

```suggestion
        "robots.txt",
```

Comment (MysteryManav) [reply api/conf/urls/__init__.py:39]: ok will keep that in mind

**Review by obulat — APPROVED** [review]
Thank you for your contribution, @MysteryManav! 🎉

**Pull request merged**